File: C++/Medium/WrongAnswer/number-of-subarrays-with-lcm-equal-to-k_2.cpp
class Solution {
public:
    int subarrayLCM(vector<int>& nums, int k) {
        
        int n = nums.size();
        vector<int> dp(n, 0);
        dp[0] = nums[0];
        for (int i = 1; i < n; ++i) {
            dp[i] = gcd(dp[i-1], nums[i]);
        }
        int res = 0;
        for (int i = 0; i < n; ++i) {
            if (dp[i] == k) {
                ++res;
            }
            for (int j = i+1; j < n; ++j) {
                int lcm = dp[j] / gcd(dp[j], nums[i]);
                if (lcm == k) {
                    ++res;
                }
            }
        }
        return res;
    }
    int gcd(int a, int b) {
        if (a < b) {
            return gcd(b, a);
        }
        if (a % b == 0) {
            return b;
        }
        return gcd(b, a%b);
    }
};

File: Java/Medium/CompileError/number-of-subarrays-with-lcm-equal-to-k_2.java
class Solution {
    public int subarrayLCM(int[] nums, int k) {
        
        int n = nums.length;
        int[] pre = new int[n + 1];
        for (int i = 0; i < n; i++) {
            pre[i + 1] = gcd(pre[i], nums[i]);
        }
        int[] count = new int[k];
        for (int i = 1; i <= n; i++) {
            int g = pre[i];
            if (g > 1) {
                count[g]++;
            }
            for (int j = 2; j * j <= g; j++) {
                if (g % j == 0) {
                    count[j]++;
                    if (j * j != g) {
                        count[g / j]++;
                    }
                }
            }
        }
        int ans = 0;
        for (int i = 1; i < k; i++) {
            if (k % i == 0) {
                ans += count[i];
            }
        }
        return ans;
    }
}